packages/frontend-plugin-api/src/wiring/createExtension.ts

```diff
@@ -512,5 +510,5 @@ export function createExtension<
         },
       }) as ExtensionDefinition<any>;
     },
-  } as InternalExtensionDefinition<T>;
+  }) as ExtensionDefinition<T>;
```

Comment:
Is the cast still needed?

Reply:
Yep, due to the `T` here not matching the default. It's not always needed though, and the earlier version of `createInstance` that didn't have the `TVersion` was able to avoid this cast. Still feels worth exploring if we can get rid of this though tbh, even if we end up with something like `.forVersion('v1').create(...)`

Reply:
Realized I could add back the type parameter for inference, was thinking about it the wrong war around. Was able to get rid of the case with that: https://github.com/backstage/backstage/pull/26624/commits/b93b1f6e4ed88ec44d03189096c0f0f75a88b505